internals/resource-managers/output.go

```diff
@@ -0,0 +1,133 @@
+)
+
+type OutputManager struct {
+	Ctx       context.Context
+	K8sclient client.Client
+	Log       logr.Logger
+}
+
+func (o OutputManager) prepareResource(capp rcsv1alpha1.Capp) loggingv1beta1.Output {
+	outputName := capp.GetName() + "-output"
+	if capp.Spec.LogSpec.Type == "elastic" {
+		output := loggingv1beta1.Output{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      outputName,
+				Namespace: capp.GetNamespace(),
+			},
+			Spec: loggingv1beta1.OutputSpec{
+				ElasticsearchOutput: &output.ElasticsearchOutput{
+					Host:       capp.Spec.LogSpec.Host,
+					Port:       9200,
+					Scheme:     "https",
+					SslVerify:  &capp.Spec.LogSpec.SSLVerify,
+					SslVersion: "TLSv1_2",
+					User:       capp.Spec.LogSpec.UserName,
+					Password: &secret.Secret{
+						ValueFrom: &secret.ValueFrom{
+							SecretKeyRef: &v1.SecretKeySelector{
+								LocalObjectReference: v1.LocalObjectReference{Name: capp.Spec.LogSpec.PasswordSecretName},
+								Key:                  "elastic",
+							},
+						},
+					},
+					IndexName: capp.Spec.LogSpec.Index,
+					Buffer: &output.Buffer{
+						Timekey:       "1m",
+						TimekeyWait:   "30s",
+						TimekeyUseUtc: true,
+					},
+				},
+			},
+		}
+		return output
+	} else if capp.Spec.LogSpec.Type == "splunk" {
+		protocol := "http"
+		if capp.Spec.LogSpec.SSLVerify == true {
+			protocol = "https"
+		}
+		insecureSSL := !capp.Spec.LogSpec.SSLVerify
+		output := loggingv1beta1.Output{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      outputName,
+				Namespace: capp.GetNamespace(),
+			},
+			Spec: loggingv1beta1.OutputSpec{
+				SplunkHecOutput: &output.SplunkHecOutput{
+					HecHost:     capp.Spec.LogSpec.Host,
+					HecPort:     8088,
+					InsecureSSL: &insecureSSL,
+					HecToken: &secret.Secret{
+						ValueFrom: &secret.ValueFrom{
+							SecretKeyRef: &v1.SecretKeySelector{
+								LocalObjectReference: v1.LocalObjectReference{Name: capp.Spec.LogSpec.HecToken},
+								Key:                  "SplunkHecToken",
+							},
+						},
+					},
+					Index:    capp.Spec.LogSpec.Index,
+					Protocol: protocol,
+					Format:   &output.Format{Type: "json"},
+					Buffer: &output.Buffer{
+						Timekey:       "1m",
+						TimekeyWait:   "30s",
+						TimekeyUseUtc: true,
+					},
+				},
+			},
+		}
+		return output
+	}
+	return loggingv1beta1.Output{}
+}
+
+func (o OutputManager) CleanUp(capp rcsv1alpha1.Capp) error {
+	outputName := capp.GetName() + "-output"
+	resourceManager := rclient.ResourceBaseManager{Ctx: o.Ctx, K8sclient: o.K8sclient, Log: o.Log}
+	output := loggingv1beta1.Output{}
+	if err := resourceManager.DeleteResource(&output, outputName, capp.Namespace); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o OutputManager) CreateOrUpdateObject(capp rcsv1alpha1.Capp) error {
+	outputName := capp.GetName() + "-output"
+	if capp.Spec.LogSpec.Type == "elastic" || capp.Spec.LogSpec.Type == "splunk" {
+		outputFromCapp := o.prepareResource(capp)
+		output := loggingv1beta1.Output{}
```

Comment:
This function misses a lot of logging , functionality. you can look on the status addon agent controller for references